### PR TITLE
New Setter Style

### PR DESCRIPTION
**Pull request by ProlificLab**
http://emberjs.com/deprecations/v1.x/#toc_computed-properties-with-a-shared-getter-and-setter

### CODE REVIEW

**Comment by dgeb** [comment]
Thanks! Will merge once Ember 1.12 is released.

**Comment by pixelhandler** [comment]
@ProlificLab @dgeb I update the bower dependencies for Ember.js 1.12 and Orbit.js 0.6.9 and the tests pass. See this commit, aside from the rebase, the only difference is the bower file https://github.com/pixelhandler/ember-orbit/commit/442185b471b5ae7eef3aa2965df9c600a72bec31

I rebased on master, added the above commit and created a PR on the @ProlificLab fork: https://github.com/ProlificLab/ember-orbit/pull/1

File: lib/ember-orbit/fields/attr.js [hunk 1]
```diff
@@ -11,8 +11,10 @@ var attr = function(type, options) {
     isAttribute: true
   };
 
-  return Ember.computed(function(key, value) {
-    if (arguments.length > 1) {
+  return Ember.computed({
+    get: function(key, value) {
```
Comment (gnarf) [inline lib/ember-orbit/fields/attr.js]: value is never used -- do we have this flag turned on for jshint?

File: lib/ember-orbit/fields/has-one.js [hunk 2]
```diff
@@ -15,11 +15,13 @@ var hasOne = function(model, options) {
     isLink: true
   };
 
-  return Ember.computed(function(key, value) {
-    var proxy = this.getLink(key);
-
-    if (arguments.length > 1) {
-      var currentValue = get(proxy, 'content');
+  return Ember.computed({
+    get: function(key, value) {
```
Comment (gnarf) [inline lib/ember-orbit/fields/has-one.js]: same ^ `value` unused

File: lib/ember-orbit/fields/key.js [hunk 3]
```diff
@@ -16,8 +16,11 @@ var key = function(type, options) {
     isKey: true
   };
 
-  return Ember.computed(function(name, value) {
-    if (arguments.length > 1) {
+  return Ember.computed({
+    get: function(name, value) {
```
Comment (gnarf) [inline lib/ember-orbit/fields/key.js]: same ^ `value` unused

File: lib/ember-orbit/schema.js [hunk 4]
```diff
@@ -11,18 +11,26 @@ var get = Ember.get;
...
+      if (!this[_property]) {
+        this[_property] = defaultValue;
+      }
+      return this[_property];
     }
-    if (!this[_property]) {
-      this[_property] = defaultValue;
-    }
-    return this[_property];
-  }.property();
```
Comment (gnarf) [inline lib/ember-orbit/schema.js]: I feel like we should set `EmberENV.EXTEND_PROTOTYPES = false` in our tests

**Comment by gnarf** [comment]
*bump* we should get this in. I'm trying to update to Ember 1.12 and would like to get rid of these deprecations

File: lib/ember-orbit/schema.js (hunk 4 above)
Comment (dgeb) [inline lib/ember-orbit/schema.js]: @gnarf agreed

**Comment by dgeb** [comment]
@gnarf sure, let me get this merged

**Comment by gnarf** [comment]
@dgeb I'm updating with some other changes too at the moment... enabling the extend prototypes false

**Comment by dgeb** [comment]
@gnarf ok - great - I'll wait for your PR then.

**Comment by gnarf** [comment]
I made these changes and got it landed - thanks for the great base to work from @ProlificLab

**Pull request merged**